drivers/sensor/ina23x/ina237.c

```diff
@@ -20,51 +20,69 @@ LOG_MODULE_REGISTER(INA237, CONFIG_SENSOR_LOG_LEVEL);
 #define INA237_CAL_SCALING 8192ULL
 
 /** @brief The LSB value for the bus voltage register, in microvolts/LSB. */
-#define INA237_BUS_VOLTAGE_UV_LSB 3125
+#define INA237_BUS_VOLTAGE_TO_uV(x) ((x) * 3125U)
 
-/** @brief Power scaling (scaled by 10) */
-#define INA237_POWER_SCALING 2
+/** @brief Power scaling (need factor of 0.2) */
+#define INA237_POWER_TO_uW(x) ((x) / 5ULL)
+
+/**
+ * @brief Scale die temperture from 0.125 degC/bit to micro-degrees C
+ *  Note that the bottom 4 bits are reserved and are always zero.
+ */
+#define INA237_DIETEMP_TO_uDegC(x) (((x) >> 4) * 125000)
+
+static void micro_s32_to_sensor_value(struct sensor_value *val, int32_t value_microX)
+{
+	val->val1 = value_microX / 1000000L;
+	val->val2 = abs(value_microX) % 1000000L;
+	if (value_microX < 0) {
+		val->val2 = -val->val2;
+	}
+}
+
+static void micro_u64_to_sensor_value(struct sensor_value *val, uint64_t value_microX)
+{
+	val->val1 = value_microX / 1000000U;
+	val->val2 = value_microX % 1000000U;
+	if (value_microX < 0) {
```

Comment:
*unsigned* 64-bit int `value_microX` is unlikely to be less than zero.

Reply:
Ooops, a bit of late night-copy-and-paste didn't go over well.  Good catch, will remove the check.